app/src/components/common/GuestApplicationContext.tsx

```diff
@@ -0,0 +1,134 @@
+import {Box, Stack} from '@mui/material';
+import {Formik} from 'formik';
+import {useEffect, useState} from 'react';
+import {Outlet} from 'react-router-dom';
+import ProgressBar from './ProgressBar';
+import {GuestApplicationSchema} from '../../utils/GuestApplicationSchema';
+import {NavButtons} from './GuestApplicationButtons';
+import {useNavigate} from 'react-router-dom';
+
+export interface formInputValues {
+  fullName: string;
+  dateOfBirth: string;
+  gender: string;
+  email: string;
+  phoneNumber: string;
+  contactPrefrence: string;
+  nameOfGuest: Array<string>;
+  typeOfPet: Array<string>;
+  currentlyEmployed: string;
+  employmentSearch: string;
+  employmentDescription: string;
+  educationEnrollment: string;
+  educationEnrollmentSearch: string;
+  educationProgramName: string;
+  multilingual: string;
+  spokenLanguages: string;
+  cigarettesUse: string;
+  alcoholUse: string;
+  otherSubstanceUse: string;
+  mentalIllness: string;
+  programGoals: string;
+  hostRelationshipGoal: string;
+  potentialChallenges: string;
+}
+export const stepToRouteMapping: {[key: number]: string} = {
+  1: '',
+};
+export const initialValues = {
+  fullName: '',
+  dateOfBirth: '',
+  gender: '',
+  email: '',
+  phoneNumber: '',
+  contactPrefrence: '',
+  nameOfGuest: [''],
+  typeOfPet: [''],
+  currentlyEmployed: '',
+  employmentSearch: '',
+  employmentDescription: '',
+  educationEnrollment: '',
+  educationEnrollmentSearch: '',
+  educationProgramName: '',
+  multilingual: '',
+  spokenLanguages: '',
+  cigarettesUse: '',
+  alcoholUse: '',
+  otherSubstanceUse: '',
+  mentalIllness: '',
+  programGoals: '',
+  hostRelationshipGoal: '',
+  potentialChallenges: '',
+};
+
+export const GuestApplicationContext = () => {
+  const TOTALPAGES = 10; /*Once all forms are made and added to "stepToRouteMapping", make this number change based on "stepToRouteMapping"*/
+  const storedStep = localStorage.getItem('currentStep');
+  const initialStep = storedStep ? parseInt(storedStep, 10) : 1;
+  const [step, setStep] = useState<number>(initialStep);
```

Comment:
I think we can place lines 66 and 67 in the initializer function for the step state like this:
```
const [step, setStep] = useState<number>(() => {
   const storedStep = localStorage.getItem('currentStep');
   const initialStep = storedStep ? parseInt(storedStep, 10) : 1;
   return initialStep;
});
```
The way it's currently setup we're calculating the initial state and getting the step from local storage on every render. Here is some info from the [React docs](https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state)

Reply:
Ahh, I see that I was initially accessing localStorage every render instead of only when it was needed which is the initial Render. Nice catch and I learned something new!